tools/setup.py

```diff
@@ -0,0 +1,55 @@
+#!/bin/python3
+# Copyright 2023 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import date
+from glob import glob
+from setuptools import setup
+import os
+
+version = '0.0.1.dev' + date.today().strftime("%Y%m%d")
+project_name = 'gematria'
+
+SO_DIRECTORIES = ['./pybind11_abseil', './gematria']
```

Comment:
We should be careful about relative paths. The current path of the script might not be something we control, so it would be better to either go relative to `__file__`, or, perhaps even better, use some other mechanism for pulling files out of Bazel `data` targets, e.g. https://github.com/bazelbuild/bazel/blob/master/tools/python/runfiles/runfiles.py.

Reply:
I ended up just using a `genrule` that creates a manifest of shared objects using the `rootpath <target>` make variable which, as I understand it, gets the exact path relative to the CWD of the binary that it can use to access the file. I don't believe this supports cross platform builds by default (since Windows by default doesn't use runfiles), but I don't think Windows support is a large priority (and this would be fun if we invoked bazel with `--enable_runfiles`), and doing it with `rlocationpath` confuses `setuptools` a lot since it gives paths not in the `.runfiles` directory.